bin/logstash.lib.sh

```diff
@@ -15,19 +15,34 @@ setup_java() {
   fi
 
   if [ ! -x "$JAVACMD" ] ; then
-    echo "Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME."
+    echo "Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME." 1>&2
     exit 1
   fi
 
-  JAVA_OPTS="$JAVA_OPTS -Xmx${LS_HEAP_SIZE}"
-  JAVA_OPTS="$JAVA_OPTS -XX:+UseParNewGC"
-  JAVA_OPTS="$JAVA_OPTS -XX:+UseConcMarkSweepGC"
-  JAVA_OPTS="$JAVA_OPTS -Djava.awt.headless=true"
+  if [ "$JAVA_OPTS" ] ; then
+    echo "WARNING: Default JAVA_OPTS will be overridden by the JAVA_OPTS defined in the environment. Environment JAVA_OPTS are $JAVA_OPTS"  1>&2
+  else
+    # There are no JAVA_OPTS set from the client, we set a predefined
+    # set of options that think are good in general
+    JAVA_OPTS="-XX:+UseParNewGC"
+    JAVA_OPTS="$JAVA_OPTS -XX:+UseConcMarkSweepGC"
+    JAVA_OPTS="$JAVA_OPTS -Djava.awt.headless=true"
+
+    JAVA_OPTS="$JAVA_OPTS -XX:CMSInitiatingOccupancyFraction=75"
+    JAVA_OPTS="$JAVA_OPTS -XX:+UseCMSInitiatingOccupancyOnly"
+  fi
 
-  JAVA_OPTS="$JAVA_OPTS -XX:CMSInitiatingOccupancyFraction=75"
-  JAVA_OPTS="$JAVA_OPTS -XX:+UseCMSInitiatingOccupancyOnly"
+  if [ "$LS_JAVA_OPTS" ] ; then
+    # The client set the variable LS_JAVA_OPTS, choosing his own
+    # set of java opts.
+    JAVA_OPTS="$JAVA_OPTS $LS_JAVA_OPTS"
+  fi
+
+  if [ "$LS_HEAP_SIZE" ] ; then
+    JAVA_OPTS="$JAVA_OPTS -Xmx${LS_HEAP_SIZE}"
+  fi
 
-  if [ ! -z "$LS_USE_GC_LOGGING" ] ; then
+  if [ "$LS_USE_GC_LOGGING" ] ; then
```

Comment:
I had a fear that this kind of checking for an empty string would only work in bash's bourne-compatibility mode (/bin/sh linked to bash). 

Just to check, I tried this under both dash (debian) and bash (osx, fedora, etc): `/bin/sh -c '[ "" ] && echo "OK"'`. All systems I tested worked fine (a few Linux variants like Debian & Ubuntu; OSX works also).

(I'm mostly just noting this for posterity)